waltz-tools/src/main/java/com/wepay/waltz/tools/cluster/ClusterCli.java

```diff
@@ -130,6 +139,361 @@ protected String getUsage() {
+         */
+        private void buildZookeeperPartitionAssignmentsValidation(ZooKeeperClient zkClient,
+                                                                  String zkRoot,
+                                                                  List<PartitionValidationResults> partitionValidationResultsList) throws ClusterManagerException {
+
+            ClusterManager clusterManager = new ClusterManagerImpl(zkClient, new ZNode(zkRoot), partitionAssignmentPolicy);
+            PartitionAssignment partitionAssignment = clusterManager.partitionAssignment();
+            int[] partitionToServerMap = new int[clusterManager.numPartitions()];
+
+            for (int serverId : partitionAssignment.serverIds()) {
+                for (PartitionInfo partitionInfo : partitionAssignment.partitionsFor(serverId)) {
+                    String error = "";
+
+                    if (partitionInfo.partitionId < 0 || partitionInfo.partitionId >= clusterManager.numPartitions()) {
+                        error = "Error: Server " + serverId + " handles invalid partition " + partitionInfo.partitionId;
+                    } else if (partitionToServerMap[partitionInfo.partitionId] != 0) {
+                        error = "Error: Partition handled by more than one server: "
+                                + partitionToServerMap[partitionInfo.partitionId] + " and " + serverId;
+                    } else {
+                        partitionToServerMap[partitionInfo.partitionId] = serverId;
+                    }
+
+                    ValidationResults validationResults = new ValidationResults(
+                                ValidationResults.ValidationType.PARTITION_ASSIGNMENT_ZK_VALIDITY,
+                                ("".equals(error)) ? ValidationResults.Status.SUCCESS : ValidationResults.Status.FAILURE,
+                                error);
+                    PartitionValidationResults partitionValidationResults = partitionValidationResultsList.get(partitionInfo.partitionId);
+                    partitionValidationResults.validationResultsMap.put(validationResults.type, validationResults);
+                }
+            }
+
+            for (int partitionId = 0; partitionId < clusterManager.numPartitions(); partitionId++) {
+                if (partitionToServerMap[partitionId] == 0) {
+                    ValidationResults validationResults = new ValidationResults(
+                            ValidationResults.ValidationType.PARTITION_ASSIGNMENT_ZK_VALIDITY,
+                            ValidationResults.Status.FAILURE,
+                            "Error: Partition " + partitionId + " not handled by any server");
+                    PartitionValidationResults partitionValidationResults = partitionValidationResultsList.get(partitionId);
+                    partitionValidationResults.validationResultsMap.put(validationResults.type, validationResults);
+                    partitionValidationResultsList.add(partitionValidationResults);
```

Comment:
what is the intention of this `partitionValidationResultsList.add` ?? won't this add to the end of the `partitionValidationResultsList` ?? whereas the size of `partitionValidationResultsList` should be just clusterManager.numPartitions...
is this statement required??

Reply:
Oops. Will fix, thanks!

Reply:
fixed. Thanks!